demos/multi_material/compositional_buoyancy/compositional_buoyancy.py

```diff
@@ -197,96 +211,70 @@ def cosine_curve(x, amplitude, wavelength, vertical_shift):
 
 # +
 output_file = VTKFile("output.pvd")
+output_file.write(*z.subfunctions, psi, time=time_now)
 
 plog = ParameterLog("params.log", mesh)
 plog.log_str("step time dt u_rms entrainment")
 
 gd = GeodynamicalDiagnostics(z, T, bottom_id, top_id)
 
-material_area = material_interface_y * lx  # Area of tracked material in the domain
+material_area = initial_interface_y * lx  # Area of tracked material in the domain
 entrainment_height = 0.2  # Height above which entrainment diagnostic is calculated
 # -
 
-# Here, we set up the variational problem for the Stokes and level-set systems. The
-# former depends on the approximation defined above, and the latter includes both
-# advection and reinitialisation components. Subcycling is available for level-set
-# advection and is mainly useful when the problem at hand involves multiple CFL
-# conditions, with the CFL for level-set advection being the most restrictive.
-
-# +
-stokes_solver = StokesSolver(
-    z,
-    T,
-    approximation,
-    bcs=stokes_bcs,
-    nullspace=Z_nullspace,
-    transpose_nullspace=Z_nullspace,
-)
-
-subcycles = 1  # Number of advection solves to perform within one time step
-level_set_solver = LevelSetSolver(psi, u, delta_t, eSSPRKs10p3, subcycles, epsilon)
-# Increase the reinitialisation time step to make up for the coarseness of the mesh
-level_set_solver.reini_params["tstep"] *= 20
-# -
-
 # Finally, we initiate the time loop, which runs until the simulation end time is
 # attained.
 
 # +
-step = 0  # A counter to keep track of looping
-output_counter = 0  # A counter to keep track of outputting
+step = 0  # A counter to keep track of loop iterations
+output_counter = 1  # A counter to keep track of outputting
 time_end = 2000
 while True:
-    # Write output
-    if time_now >= output_counter * output_frequency:
-        output_file.write(*z.subfunctions, T, psi)
-        output_counter += 1
-
     # Update timestep
-    if time_end is not None:
-        t_adapt.maximum_timestep = min(output_frequency, time_end - time_now)
+    if time_end - time_now < output_frequency:
+        t_adapt.maximum_timestep = time_end - time_now
```

Comment:
Why adjusting `t_adapt` instead of directly `delta_t`?
```suggestion
delta_t.assign(time_end - time_now)
```

Reply:
That could make the solver crash if `output_frequency` is much larger than what is allowed by the CFL criterion.